osu.Game/Screens/Edit/Compose/Components/SelectionHandler.cs

```diff
@@ -145,10 +145,16 @@ public void OnReleased(PlatformAction action)
         /// <param name="blueprint">The blueprint.</param>
         internal void HandleSelected(SelectionBlueprint blueprint)
         {
-            selectedBlueprints.Add(blueprint);
-            EditorBeatmap.SelectedHitObjects.Add(blueprint.HitObject);
+            if (!selectedBlueprints.Contains(blueprint))
```

Comment:
Isn't this guaranteed to be false? The only way this can happen is if `onBlueprintSelected()` is called multiple times, which seems to be impossible due to the implementation of `SelectionBlueprint.State` (where it avoids multiple events if already selected).

I.e. this code could be reduced to just
```
selectedBlueprints.Add(blueprint);

if (!EditorBeatmap.SelectedHitObjects.Contains(...)
    ...

UpdateVisibility();
```

Reply:
sure, can be removed.